src/Symfony/Component/Translation/DependencyInjection/TranslatorPass.php

```diff
@@ -55,8 +55,13 @@ public function process(ContainerBuilder $container)
 
             foreach ($container->findTaggedServiceIds('validator.constraint_validator', true) as $id => $attributes) {
                 $serviceDefinition = $container->getDefinition($id);
+                $className = $serviceDefinition->getClass();
+                // Resolve Constraint Validator FQCN defined as %foo.bar.class% parameter
+                if (str_starts_with($className, '%')) {
```

Comment:
A parameter does not necessarily have to be located at the beginning of the string, it could be anywhere.

Again, this compiler pass should not need to know how a parameter looks like at all. Just call that `resolveValue()` method unconditionally and if there's noting to resolve, it'll be a no-op.

Reply:
Yes, you're right. I do not need IF there.